pulp_rpm/tests/functional/conftest.py

```diff
@@ -109,6 +111,64 @@ def signed_artifact(http_get, artifacts_api_client, gen_object_with_cleanup, tmp
     return gen_object_with_cleanup(artifacts_api_client, temp_file).to_dict()
 
 
+@pytest.fixture
+def rpm_artifact_factory(
+    http_get, artifacts_api_client, gen_object_with_cleanup, tmp_path, pulp_domain_enabled
+):
+    """Return an artifact created from uploading an RPM file."""
+
+    def _rpm_artifact_factory(url=RPM_SIGNED_URL, pulp_domain=None):
+        temp_file = tmp_path / str(uuid.uuid4())
+        temp_file.write_bytes(http_get(url))
+        kwargs = {}
+        if pulp_domain:
+            if not pulp_domain_enabled:
+                raise RuntimeError("Server does not have domains enabled.")
+            kwargs["pulp_domain"] = pulp_domain
+        return gen_object_with_cleanup(artifacts_api_client, temp_file, **kwargs).to_dict()
```

Comment:
Does this need to have `to_dict()` at the end? I think some of the other factories have it, but I would prefer if we keep the original object returned.